From fdaf90c50c0efb7fa4c41ca5d672f112e6729017 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Wed, 12 Apr 2023 20:05:08 +0000
Subject: [PATCH 11/12] WIP [compiler-rt] [asan] Fix building tests for
 mingw/i386

In the case of mingw, the CRT is always dynamically linked, so the
!ASAN_DYNAMIC case isn't really relevant. The symbol referenced,
_except_handler4, isn't available with the dynamically linked CRT
(for ASAN_DYNAMIC, the symbol is redirected to _except_handler4_common).
---
 compiler-rt/lib/asan/asan_win.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compiler-rt/lib/asan/asan_win.cpp b/compiler-rt/lib/asan/asan_win.cpp
index 7dbd7ab98a17..c48ef77e48c1 100644
--- a/compiler-rt/lib/asan/asan_win.cpp
+++ b/compiler-rt/lib/asan/asan_win.cpp
@@ -173,7 +173,9 @@ void InitializePlatformInterceptors() {
   ASAN_INTERCEPT_FUNC(__C_specific_handler);
 #else
   ASAN_INTERCEPT_FUNC(_except_handler3);
+#if ASAN_DYNAMIC || !defined(__MINGW32__)
   ASAN_INTERCEPT_FUNC(_except_handler4);
+#endif
 #endif
 
   // Try to intercept kernel32!RaiseException, and if that fails, intercept
-- 
2.34.1

